daemon/containerd/image_snapshot.go

```diff
@@ -63,17 +65,21 @@ func (i *ImageService) PrepareSnapshot(ctx context.Context, id string, parentIma
 	if err != nil {
 		return err
 	}
-
 	if err := ls.AddResource(ctx, lease, leases.Resource{
 		ID:   id,
 		Type: "snapshots/" + i.StorageDriver(),
 	}); err != nil {
 		return err
 	}
 
-	s := i.client.SnapshotService(i.StorageDriver())
-	_, err = s.Prepare(ctx, id, parentSnapshot)
-	return err
+	snapshotter := i.client.SnapshotService(i.StorageDriver())
+
+	if i.idMapping.Empty() {
+		_, err = snapshotter.Prepare(ctx, id, parentSnapshot)
+		return err
+	}
+
+	return i.remapSnapshot(ctx, snapshotter, id, parentSnapshot, lease)
```

Comment:
Ohman, changes look good, but I'm just looking at this part again, and this _could be_ a potential foot-gun if either a linter, or some "helpful" contribution fixes "unhandled error" here;

```suggestion

	if i.idMapping.Empty() {
		if _, err := snapshotter.Prepare(ctx, id, parentSnapshot); err != nil {
			return err
		}
	}

	return i.remapSnapshot(ctx, snapshotter, id, parentSnapshot, lease)
```


The above would break this code (which would _hopefully_ be caught in CI.

Do you think it's worth to be defensive here, and to reverse the order?

```suggestion

	if !i.idMapping.Empty() {
		return i.remapSnapshot(ctx, snapshotter, id, parentSnapshot, lease)
	}

	_, err = snapshotter.Prepare(ctx, id, parentSnapshot)
	return err
```

Reply:
Sure I can reverse the order :+1: